### PR TITLE
Fixed tests for staticmethod

### PR DESCRIPTION
**Pull request by da-woods**
This is intended to fix the staticmethod tests when `binding=True` (https://github.com/cython/cython/pull/2864). One of the tests will fail if run against the current master so it should only be applied to the enable_binding branch

The upshot is that Cython was previously doing the wrong thing for Python 2. The test

```
def f_plus(a):
    return a + 1

class class1:
    plus1 = f_plus
```

should fail with a `TypeError` for Python 2. The test now compares against pure Python behaviour. It doesn't (currently) test with `binding=False` - I can add this test if useful.

### CODE REVIEW

File: tests/run/staticmethod_pure.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+def f_plus(a):
+    return a + 1
+
+class class1:
+    plus1 = f_plus
+
+class class2(object):
+    plus1 = f_plus
```
Comment (scoder) [inline tests/run/staticmethod_pure.py]: These examples do not use static methods at all, the use entirely normal methods. I can see that this test was previously in `staticmethod.pyx`, but moving it out calls for a better name of this test file now. Maybe `method_assignment.py` ?

Comment (da-woods) [reply tests/run/staticmethod_pure.py]: OK - I've made that change. The other option might be to put them in with the function_as_method_T494 py and pyx file which cover a similar (but not quite identical) theme. It'd also provide somewhere to put `class3`.

Comment (da-woods) [reply tests/run/staticmethod_pure.py]: I've moved them there. While doing that I've done a bit of tidying up of the class names, and added a test for the old `binding(False)` behaviour (which I assume will continue to exist if specifically requested)

**Comment by scoder** [comment]
Thanks!

**Pull request merged**